reactor/aws/arn_utils.py

```
@@ -0,0 +1,301 @@
+    'es': 'subscription',
+    'event-source-mappings': 'event-source-mapping',
+    'hostedzone': 'hosted-zone',
+    'inputsecuritygroup': 'input-security-group',
+    'ipset': 'ip-set',
+    'jobs': 'job',
+    'jobtemplates': 'job-template',
+    'maintenancewindow': 'maintenance-window',
+    'mlmodel': 'learning-model',
+    'og': 'option-group',
+    'origin_endpoints': 'origin-endpoint',
+    'ou': 'organizational-unit',
+    'parametergroup': 'parameter-group',
+    'patchbaseline': 'baseline',
+    'pg': 'parameter-group',
+    'presets': 'preset',
+    'queues': 'queue',
+    'scalingpolicy': 'policy',
+    'secgrp': 'security-group',
+    'securitygroup': 'security-group',
+    'solutionstack': 'solution-stack',
+    'statemachine': 'state-machine',
+    'subgrp': 'subnet-group',
+    'subnetgroup': 'subnet-group',
+    'threatintelset': 'threat-intel-set',
+    'vaults': 'vault',
+    'webacl': 'web-acl',
+}
+
+
+def get_resource_type_from_arn(arn):
+    """
+    Get a normalized resource type from an ARN
+
+    Args:
+        arn: (str) - the arn to deconstruct
+
+    Returns:
+        str - a canonical resource type name
+    """
+    if not arn:
+        return None
+    try:
+        arn_match = re.fullmatch(_ARN_MATCH_REGEX, arn)
+        if not arn_match or len(arn_match.groups()) != 4:
+            raise ValueError('ARN did not match standard regex')
+
+        service, _, _, resource = arn_match.groups()
+        if service not in _service_types_in_arns:
+            raise ValueError(f'Unknown service {service}')
+        custom_handlers_map = _get_resource_type_custom_handlers()
+        if service in custom_handlers_map:
+            return custom_handlers_map[service](resource)
+        elif '/' in resource:
+            return _normalize_resource_type_name(resource.split('/')[0])
+        elif ':' in resource:
+            return _normalize_resource_type_name(resource.split(':')[0])
+        else:
+            return None
```

should this raise a value error instead of returning None? If we can't extract a resource type thats a but or at the very lest something we need to know

---

I could go either way with this - I think the decision is going to depend on whether or not the caller is equipped to deal with a raised exception.  If you think the answer is "yes" I'll go ahead and change this

---

yeah per discussion, it's fine as is